### DISCUSSION TITLE
Fido2 Webauthn broken on latest version of Android app (2.13.0)

### DISCUSSION THREAD

**Discussion by krzmaz** [dani-garcia/vaultwarden#1997] [Q&A] [answered]
### Subject of the issue

It seems that with the recent version of the mobile app some changes were introduced to FIDO2 Webauthn authentication - https://github.com/bitwarden/mobile/pull/1519 
After updating, there is a 404 error when trying to authenticate, as the app is trying to connect to an unimplemented endpoint, example below:  
https://vault.mydomain.com/webauthn-mobile-connector.html?data=[DATA]=&parent=bitwarden%3A%2F%2Fwebauthn-callback&v=2 

### Deployment environment

* vaultwarden version: 1.22.2

* Install method: docker image

* Clients used:  Android

* Reverse proxy and version: 

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:

### Steps to reproduce

Try to authenticate with Fido2 Webauthn using the latest (2.13.0) mobile app

### Expected behaviour

Authentication is possible

### Actual behaviour

Error 404 happens

**Comment by BlackDex** [comment]
I think it should work fine. Are you using a reverse proxy, and if so which one? Does it has caching functionally? Did you clear the cache if so?

**Comment by BlackDex** [comment, marked as answer]
Sorry, i just saw that this specific endpoint is available within the current testing image, and thus already fixed. But not in the current latest version 

Either use the testing image or wait for a new release.

**Comment by krzmaz** [comment]
Thank you for the quick response!
The latest testing docker image is panicking and not starting on my Raspberry Pi 4, so I guess I'll wait for a release :)
Thank you for all the work on this project!

**Reply by BlackDex** [reply]
@TheRealMazur  Could  you provide the panic? Because that should not happen, and a new release will be based upon the `testing` images.

**Reply by krzmaz** [reply]
This is what I did when I saw that the `testing` container isn't starting:
```
~/vw-data $ docker run -i --name vaultwarden -p 8090:80 vaultwarden/server:testing
/--------------------------------------------------------------------\
|                        Starting Vaultwarden                        |
|                      Version 1.22.2-b4c95fb4                       |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://vaultwarden.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/vaultwarden/issues/new            |
\--------------------------------------------------------------------/

[INFO] No .env file found.

thread panicked while processing panic. aborting.
~/vw-data $
```
I haven't investigated it further. How can I enable more verbose output?

**Reply by BlackDex** [reply]
Could you please run it like this? I hope that we see some more info:
```bash
docker run --rm -it --name vaultwarden -e RUST_BACKTRACE=1 -e LOG_LEVEL=trace -p 8090:80 vaultwarden/server:testing
```

And else like this to have some docker debugging also:
```bash
docker --debug run --rm -it --name vaultwarden -e RUST_BACKTRACE=1 -e LOG_LEVEL=trace -p 8090:80 vaultwarden/server:testing
```

**Reply by krzmaz** [reply]
The first command didn't provide any more info, the second one only added:
```
DEBU[0002] [hijack] End of stdout
```
at the end... :/

**Reply by BlackDex** [reply]
It looks like something goes wrong on the host.
No clue what it could be though.

What happens if you do this then?
```bash
docker --debug run --rm -it --name vaultwarden -e RUST_BACKTRACE=1 -e LOG_LEVEL=trace -p 8090:80 vaultwarden/server:latest
```
or
```bash
docker --debug run --rm -it --name vaultwarden -e RUST_BACKTRACE=1 -e LOG_LEVEL=trace -p 8090:80 vaultwarden/server:testing-alpine
```

**Reply by krzmaz** [reply]
Both of those start normally with a lot of logs. Do you want me to paste the logs?

**Reply by BlackDex** [reply]
No, if they start that is fine. Ok, i have one last try, else I'm at a loss why it doesn't want to start.
```bash
docker --debug run --rm -it --name vaultwarden -e RUST_BACKTRACE=1 -e LOG_LEVEL=trace -p 8090:80 vaultwarden/server:testing@sha256:56c50760118afc2a8006492ed51b2476ae4974444e9754b1d05153f0e2c51cc1
```

Only thing i can think of is something on the host is doing something funny.

**Reply by krzmaz** [reply]
That's the wrong platform for raspberry pi:
```
Status: Downloaded newer image for vaultwarden/server@sha256:56c50760118afc2a8006492ed51b2476ae4974444e9754b1d05153f0e2c51cc1                             WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm/v7) and no specific platform was requested
standard_init_linux.go:228: exec user process caused: exec format error
DEBU[0051] [hijack] End of stdout
DEBU[0051] Error resize: Error response from daemon: Container 9ed3201b045ba4 84484c849cf7c8e611150261fdaf99e732efb82f8e5d384a4 is not running
```

**Reply by BlackDex** [reply]
Ah, then in that case use this for armv7:
```bash
docker --debug run --rm -it --name vaultwarden -e RUST_BACKTRACE=1 -e LOG_LEVEL=trace -p 8090:80 vaultwarden/server:testing@sha256:9362f67521460310f78b524cdbe4fc8d9029aa0113d0a8ecdde9f3d0e9e0b0b5
```

**Reply by krzmaz** [reply]
No luck:
```
~/vw-data $ docker --debug run --rm -it --name vaultwarden -e RUST_BACKTRACE=1 -e LOG_LEVEL=trace -p 8090:80 vaultwarden/server:testing@sha256:9362f67521460310f78b524cdbe4fc8d9029aa0113d0a8ecdde9f3d0e9e0b0b5
/--------------------------------------------------------------------\
|                        Starting Vaultwarden                        |
|                      Version 1.22.2-b4c95fb4                       |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://vaultwarden.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/vaultwarden/issues/new            |
\--------------------------------------------------------------------/

[INFO] No .env file found.

thread panicked while processing panic. aborting.
DEBU[0003] [hijack] End of stdout
~/vw-data $
```
I've even tested older images, and had the same results:
```
/--------------------------------------------------------------------\
|                        Starting Vaultwarden                        |
|                      Version 1.22.2-d5ed2ce6                       |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://vaultwarden.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/vaultwarden/issues/new            |
\--------------------------------------------------------------------/

[INFO] No .env file found.

thread panicked while processing panic. aborting.
DEBU[0059] [hijack] End of stdout
```

I will try to do a bisect to find the failing image, but that's a quite manual process...

**Reply by krzmaz** [reply]
Update:
Image Version 1.22.2-0d2b3bfb works, so the regression was introduced somewhere between 0d2b3bfb and d5ed2ce6

**Reply by BlackDex** [reply]
I just started that same image on my rpi, and i get an error also, though it is later in the start process, and i do get a normal panic, and the rest just continues.

**Comment by BlackDex** [comment]
Could you create a new ticket regarding this panic please?
And also mention which specific rpi you have.

An output of `cat /proc/cpuinfo` in that ticket would also be nice.

**Reply by krzmaz** [reply]
Sure, created https://github.com/dani-garcia/vaultwarden/issues/1998 with the summary of my findings
